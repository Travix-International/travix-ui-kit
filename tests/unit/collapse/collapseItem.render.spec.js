import { shallow } from 'enzyme';
import React from 'react';
import CollapseItem from '../../../components/collapse/collapseItem';

describe('Collapse: render', () => {
  it('should render collapse item component', () => {
    const component = shallow(
      <CollapseItem
        id="c1"
        title="Collapse Item 1"
      >
        <div>Collapse Item content</div>
      </CollapseItem>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render expanded collapse item component', () => {
    const component = shallow(
      <CollapseItem
        id="c1"
        isActive
        title="Collapse Item 1"
      >
        <div>Collapse Item content</div>
      </CollapseItem>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render collapse item without id', () => {
    const component = shallow(
      <CollapseItem title="Collapse Item 1">
        <div>Collapse Item content</div>
      </CollapseItem>
    );
    expect(component).toMatchSnapshot();
  });

  it('should call onClick after click on label with correct id', () => {
    const e = { target: {} };
    const onClick = jest.fn();
    const component = shallow(
      <CollapseItem
        id="c1"
        onClick={onClick}
        title="Collapse Item 1"
      >
        <div>Collapse Item content</div>
      </CollapseItem>
    );
    component.find('.ui-collapse__label').simulate('click', e);
    expect(onClick).toBeCalledWith(e, 'c1');
  });

  it('should work fine after click on label when "onClick" function is not provided', () => {
    const e = { target: {} };
    const component = shallow(
      <CollapseItem
        id="c1"
        title="Collapse Item 1"
      >
        <div>Collapse Item content</div>
      </CollapseItem>
    );
    component.find('.ui-collapse__label').simulate('click', e);
  });

  it('should render collapseItem component with icon on the right', () => {
    const component = shallow(
      <CollapseItem iconPosition="right" title="Collapse Item 1">
        <div>Collapse Item content</div>
      </CollapseItem>
    );
    expect(component).toMatchSnapshot();
  });
});
