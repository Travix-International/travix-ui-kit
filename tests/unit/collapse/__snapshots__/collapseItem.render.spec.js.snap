// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse: render should render collapse item component 1`] = `
<div
  className="ui-collapse-item"
>
  <button
    aria-controls="c1"
    aria-expanded={false}
    className="ui-collapse__label ui-collapse__label--icon_position ui-collapse__label--icon_position_left"
    onClick={[Function]}
    type="button"
  >
    <span
      className="ui-collapse__label-icon"
    />
    <span
      className="ui-collapse__label-text"
    >
      Collapse Item 1
    </span>
  </button>
  <div
    aria-hidden={true}
    className="ui-collapse-item__content"
    id="c1"
  >
    <div>
      Collapse Item content
    </div>
  </div>
</div>
`;

exports[`Collapse: render should render collapse item without id 1`] = `
<div
  className="ui-collapse-item"
>
  <button
    aria-expanded={false}
    className="ui-collapse__label ui-collapse__label--icon_position ui-collapse__label--icon_position_left"
    onClick={[Function]}
    type="button"
  >
    <span
      className="ui-collapse__label-icon"
    />
    <span
      className="ui-collapse__label-text"
    >
      Collapse Item 1
    </span>
  </button>
  <div
    aria-hidden={true}
    className="ui-collapse-item__content"
  >
    <div>
      Collapse Item content
    </div>
  </div>
</div>
`;

exports[`Collapse: render should render collapseItem component with icon on the right 1`] = `
<div
  className="ui-collapse-item"
>
  <button
    aria-expanded={false}
    className="ui-collapse__label ui-collapse__label--icon_position ui-collapse__label--icon_position_right"
    onClick={[Function]}
    type="button"
  >
    <span
      className="ui-collapse__label-icon"
    />
    <span
      className="ui-collapse__label-text"
    >
      Collapse Item 1
    </span>
  </button>
  <div
    aria-hidden={true}
    className="ui-collapse-item__content"
  >
    <div>
      Collapse Item content
    </div>
  </div>
</div>
`;

exports[`Collapse: render should render expanded collapse item component 1`] = `
<div
  className="ui-collapse-item ui-collapse-item_active"
>
  <button
    aria-controls="c1"
    aria-expanded={true}
    className="ui-collapse__label ui-collapse__label--icon_position ui-collapse__label--icon_position_left"
    onClick={[Function]}
    type="button"
  >
    <span
      className="ui-collapse__label-icon"
    />
    <span
      className="ui-collapse__label-text"
    >
      Collapse Item 1
    </span>
  </button>
  <div
    aria-hidden={false}
    className="ui-collapse-item__content"
    id="c1"
  >
    <div>
      Collapse Item content
    </div>
  </div>
</div>
`;
